pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/util/TransactionStateSQLiteImpl.java

```diff
@@ -192,4 +192,32 @@ public void deleteCompletedFileRecord(String fileName) throws SQLException {
                 autoRollback.commit();
             }
         }
+
+        /**
+         * Update below details
+         *      1. Add entry into FailedFiles table for given file name and offset
+         *      2. Delete all entry from PendingFiles for given file
+         *
+         * @param fileName               file name of processed file
+         * @param beginOffset            begin offset from where file read starts
+         *
+         */
+        @Override
+        public void addFailedFileRecord(String fileName, long beginOffset) throws SQLException {
+            try (final PreparedStatement insertFailedFileStatement = connection.prepareStatement(
+                         "insert or ignore into FailedFiles (fileName, offset) values (?, ?)");
+                 final PreparedStatement deletePendingFileStatement = connection.prepareStatement(
```

Comment:
Do we really need to delete file entries from pending files? 
isValidFile(..) check called initially at the time of getDirectoryListing() ie reading file name from given directory path and we would be adding only new files to pendingFiles table which will be done later so I dont think that we are adding invalid files to pedningFiles table

Reply:
Yes, since this check is happening before, the files will not be added to Pending Files table. Also, since we're now moving files to a different folder we dont need to keep track within the database